src/cplscheme/CouplingData.cpp

```diff
@@ -139,6 +139,98 @@ void CouplingData::moveToNextWindow()
   }
 }
 
+Eigen::VectorXd CouplingData::getSerializedValues()
+{
+  int  nValues        = getSize();
+  int  nTimeSteps     = 2; // only worry about WINDOW_START and WINDOW_END for now.
+  auto serializedData = Eigen::VectorXd(nTimeSteps * nValues);
+
+  auto atBeginn = timeStepsStorage().stamples().front();
+  int  timeId   = 0;
+  PRECICE_ASSERT(atBeginn.timestamp == time::Storage::WINDOW_START, atBeginn.timestamp);
+  auto sliceBeginn = atBeginn.sample.values;
+  for (int valueId = 0; valueId < nValues; valueId++) {
+    serializedData(valueId * nTimeSteps) = sliceBeginn(valueId);
+  }
+
+  auto atEnd = timeStepsStorage().stamples().back();
+  timeId     = 1;
+  // PRECICE_ASSERT(atEnd.timestamp == time::Storage::WINDOW_END, atEnd.timestamp); // triggered during exchangeInitialData
```

Comment:
What do we do with this one?

Reply:
Fixed. c185890655af6a6a18e03c527112830d89114b15